### PR TITLE
[Test] Fix authentication creation in example project

### LINKED ISSUE #86378: [CI] CustomAuthorizationEngineTests.java:56: error: no suitable constructor found for Authentication(User,RealmRef,RealmRef)
### CI Link

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+pull-request+example-plugins/930/console

### Repro line

cd plugin/examples; ./gradlew build --include-build ../..

### Does it reproduce?

Yes

### Applicable branches

master

### Failure history

_No response_

### Failure excerpt

The example projects fails with a compile error caused by this change here: https://github.com/elastic/elasticsearch/pull/86206

This is blocking current PR checks as example pr build is failing.

the stack trace:
> Task :security-authorization-engine:compileTestJava FAILED
09:14:22 /dev/shm/elastic+elasticsearch+pull-request+example-plugins/plugins/examples/security-authorization-engine/src/test/java/org/elasticsearch/example/CustomAuthorizationEngineTests.java:56: error: no suitable constructor found for Authentication(User,RealmRef,RealmRef)
09:14:22                 new Authentication(new User("joe", new String[]{"custom_superuser"}, new User("bar", "not_superuser")),
09:14:22                 ^
09:14:22     constructor Authentication.Authentication(User,RealmRef,RealmRef,Version,AuthenticationType,Map<String,Object>) is not applicable
09:14:22       (actual and formal argument lists differ in length)
09:14:22     constructor Authentication.Authentication(StreamInput) is not applicable
09:14:22       (actual and formal argument lists differ in length)
09:14:22 /dev/shm/elastic+elasticsearch+pull-request+example-plugins/plugins/examples/security-authorization-engine/src/test/java/org/elasticsearch/example/CustomAuthorizationEngineTests.java:73: error: no suitable constructor found for Authentication(User,RealmRef,RealmRef)
09:14:22                 new Authentication(new User("joe", new String[]{"not_superuser"}, new User("bar", "custom_superuser")),
09:14:22

### PR DESCRIPTION
**Pull request by ywangd**
In #86206, we closed down Authentication constructors to favour
dedicated convenient methods for instantiation. The constructor usages
in the example project were however left out (another refactor fallout).

Relates: #86206
Resolves: #86378

### CODE REVIEW

File: plugins/examples/security-authorization-engine/src/test/java/org/elasticsearch/example/CustomAuthorizationEngineTests.java [hunk 1]
```diff
@@ -52,9 +52,9 @@ public void testAuthorizeRunAs() {
         CustomAuthorizationEngine engine = new CustomAuthorizationEngine();
         // unauthorized
         {
-            Authentication authentication =
-                new Authentication(new User("joe", new String[]{"custom_superuser"}, new User("bar", "not_superuser")),
-                    new RealmRef("test", "test", "node"), new RealmRef("test", "test", "node"));
+            Authentication authentication = Authentication
+                .newRealmAuthentication(new User("bar", "not_superuser"), new RealmRef("test", "test", "node"))
+                .runAs(new User("joe", "custom_superuser"), new RealmRef("test", "test", "node"));
```
Comment (n1v0lg) [inline plugins/examples/security-authorization-engine/src/test/java/org/elasticsearch/example/CustomAuthorizationEngineTests.java:57]: Neat!

**Review by breskeby — CHANGES_REQUESTED** [review]
Can you revert my changes to ignore this example project as part of this PR? https://github.com/elastic/elasticsearch/commit/177b0fa47ff6713552fabb84087122b829697dc9

**Comment by ywangd** [comment]
> Can you revert my changes to ignore this example project as part of this PR? [177b0fa](https://github.com/elastic/elasticsearch/commit/177b0fa47ff6713552fabb84087122b829697dc9)

Done. I was going to but your PR wasn't merged when this one got raised.

**Review by n1v0lg — APPROVED** [review]

**Review by breskeby — APPROVED** [review]
thank you!

**Pull request merged**